org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java

```diff
@@ -0,0 +1,68 @@
+package org.eclipse.lsp4xml.extensions.contentmodel.model;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FilesChangedTracker {
+
+	private static class FileChangedTracker {
+
+		private final Path file;
+		private FileTime lastModified;
+
+		public FileChangedTracker(Path file) {
+			this.file = file;
+			try {
+				lastModified = Files.getLastModifiedTime(file);
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+		}
+
+		public boolean isDirty() {
+			try {
+				FileTime currentLastMofied = Files.getLastModifiedTime(file);
+				if (!currentLastMofied.equals(lastModified)) {
+					lastModified = currentLastMofied;
+					return true;
+				}
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			return false;
+		}
+
+	}
+
+	private final List<FileChangedTracker> files;
+
+	public FilesChangedTracker() {
+		files = new ArrayList<>();
+	}
+
+	public void addFileURI(String fileURI) {

```

Comment:
you need a way to remove tracked files

Reply:
why?

Reply:
to prevent memory leaks?

Reply:
The track is very simply, it checks just the last modified of file eeach time. I don't use watcher files system.

Reply:
so what happens if the tracked file is deleted?

Reply:
Good catch! XSD Schema was stored in memory and completion continued to work. It should not. I have fixed the problem and cover that in my test.